File: client/src/api/endpoints/activity.ts
import type { AxiosInstance } from 'axios'

export interface ActivityDay {
  date: string
  count: number
  level: number
}

export interface ActivityStats {
  activeDays: number
  totalExpenses: number
  currentStreak: number
  longestStreak: number
}

export interface ActivityResponse {
  days: ActivityDay[]
  stats: ActivityStats
}

export const activityApi = {
  get: (client: AxiosInstance) => client.get<ActivityResponse>('/api/me/activity').then((res) => res.data),
}

File: server/app/schemas/activity.py
import datetime as dt

from pydantic import BaseModel


class ActivityDay(BaseModel):
    date: dt.date
    count: int
    level: int


class ActivityStats(BaseModel):
    activeDays: int
    totalExpenses: int
    currentStreak: int
    longestStreak: int


class ActivityResponse(BaseModel):
    days: list[ActivityDay]
    stats: ActivityStats
